### PR TITLE
- [readline] fix M1

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **readline/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/readline/all/conanfile.py [hunk 1]
```diff
@@ -66,6 +66,10 @@ def _configure_autotools(self):
             configure_args.append("bash_cv_wcwidth_broken=yes")
 
         self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
```
Comment (SpaceIm) [inline recipes/readline/all/conanfile.py:69]:
```suggestion
        if self.settings.arch == "armv8":
```

Comment (SpaceIm) [reply recipes/readline/all/conanfile.py:69]: or maybe tools.is_apple_os(self.settings.os), but for sure not only Macos.

Comment (SSE4) [reply recipes/readline/all/conanfile.py:69]: I can't test other configurations right now, it's out of scope of the PR

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/readline/all/conanfile.py (hunk 1 above)
Comment (theirix) [reply recipes/readline/all/conanfile.py:69]: `armv8` is not only 64-bit architecture from `settings.yml`, `armv8.3` is another one. Probably check should be for prefix `armv8` and not `armv8_32`

Comment (madebr) [reply recipes/readline/all/conanfile.py:69]: I think SSE4's initial code is correct, though I would check for the compiler (`self.settings.compiler == "apple-clang"`) instead.
The added link flags look compiler/linker specific.

**Review by jgsogo — APPROVED** [review]
Thanks! Let's fix M1 now, we can fix others later.

**Pull request merged**